src/controllers/topic.controller.ts

```diff
@@ -48,7 +49,7 @@ export class TopicController {
     }
   }
 
-  @Authorized()
+  @Authorized(Roles.Admin)
   @Put('/:id')
   async put(
```

Comment:
I think that is very important the name of the function, if the function has a descriptive name is easier to understand what it's doing... In this case maybe, `updateByUser` or something like that could be a best name.